src/replit/database/__init__.py

```diff
@@ -295,58 +307,284 @@ def __repr__(self) -> str:
+                )
+                if not toset:
+                    continue
+                try:
+                    data = json.loads(toset)
+                except json.JSONDecodeError:
+                    print("Invalid JSON data!")
+                else:
+                    if not self._is_valid_type(data):
+                        print(self._type_mismatch_msg(data))
+                        continue
 
-class ReplitDb(AsyncReplitDb):
-    """Client interface with the Replit Database."""
+                    self.db[self.key] = toset
+                    print("Wrote data to key")
+                    return data
 
-    def __getitem__(self, item: str) -> str:
-        """Retrieve a key from the database.
+    def set(self, data: JSON_TYPE) -> None:
+        """Set the value of the jsonkey.
 
         Args:
-            item (str): The key to retrieve.
+            data (JSON_TYPE): The value to set it to.
+
+        Raises:
+            TypeError: The type of the value set does not match the datatype.
+        """
+        if not self._is_valid_type(data):
+            raise TypeError(self._type_mismatch_msg(data))
+        self.db[self.key] = json.dumps(data)
+
+    def __getitem__(self, name: str) -> JSON_TYPE:
+        """Retrieve a key from the JSONKey's value if it is a dict.
+
+        Args:
+            name (str): The name to retrieve.
+
+        Raises:
+            TypeError: The dtype attribute is not dict.
+
+        Returns:
+            JSON_TYPE: The value of the key.
+        """
+        if self.dtype is not dict:
+            raise TypeError(
+                "Dictionary syntax can only be used if the datatype is dict"
+            )
+        return self.get()[name]
+
+    def read(self, name: str, default: Any = None) -> Any:
+        """Shorthand for self.get().get(name, default) if datatype is dict.
+
+        Args:
+            name (str): The name to get.
+            default (Any): The default if the key doesn't exist. Defaults to None.
+
+        Raises:
+            TypeError: The datatype is not dict.
 
         Returns:
-            str: The value of the key.
+            Any: The value read or the default.
+        """
+        if self.dtype is not dict:
+            raise TypeError("key() can only be used if the datatype is dict")
+        return self.get().get(name, default)
+
+    def __setitem__(self, name: str, value: JSON_TYPE) -> None:
+        """Sets a key inside the JSONKey's value if it is a dict.
+
+        Args:
+            name (str): The key to set.
+            value (JSON_TYPE): The value to set it to.
+
+        Raises:
+            TypeError: The dtype attribute is not dict.
         """
-        return self.get(item)
+        if self.dtype is not dict:
+            raise TypeError(
+                "Dictionary syntax can only be used if the datatype is dict"
+            )
+        data = self.get()
+        data[name] = value
+        self.set(data)
+
+
+class ReplitDb(dict):
```

Comment:
Why not inherit from AsyncReplitDb anymore?

Reply:
Whoops, that was a mistake! The sync code was restored from an old commit that didn't inherit